mne/viz/_figure.py

```diff
@@ -1854,6 +1863,54 @@ def _update_yaxis_labels(self):
                    else 'normal')
             text.set_style(sty)
 
+    def _xtick_formatter(self, xval, _=None):
+        """Change the x-axis labels."""
+        if self.mne.time_format == 'datetime':
+            import datetime
+            first_time = self.mne.inst.first_time
+            meas_date = self.mne.inst.info['meas_date']
+            seconds = int(xval + first_time)
+            # Get decimals with remainder of division by 1.
+            ms = int((xval + first_time) % 1 * 1e3)
+            if ms == 0:
+                us = int((xval + first_time) % 1 * 1e6)
+            else:
+                us = int(((xval + first_time) % 1 * 1e6) % (ms * 1e3))
+            # Add time from first_time (only seconds) and xval to meas_date
+            xdatetime = meas_date + datetime.timedelta(seconds=seconds,
+                                                       milliseconds=ms,
+                                                       microseconds=us)
+            xdtstr = xdatetime.strftime('%H:%M:%S')
+            # Crop time string for microseconds depending on duration
+            ustr = xdatetime.strftime('%f')
+            lim = max(int(np.ceil(-np.log10(self.mne.duration))) + 3, 0)
+            if lim > 0:
+                ustr = ustr[:lim].rstrip("0")
+                if len(ustr) > 0:
+                    xdtstr += '.' + ustr
+
+            return xdtstr
+        else:
+            return xval
```

Comment:
two other things that would be great to see here.

1. the ticks on the scrollbar never change, so they don't really need to get decimals added when zooming in.
2. the green text added when the vertical bar shows is often overlapping the first tick.

Reply:
How would you solve the overlapping green text issue?

Reply:
Not sure. One option is to give the green text a (semi-)opaque background; another is to put the green text below the green line, though it would still need the background in case it overlaps one of the other tick labels... Could also try just pushing it further left from its current position but I'm not sure that will suffice for the datetime case

Reply:
Wow nice @drammock , I didn't know that you could pass float-seconds to `datetime.timedelta()`.

Reply:
Concerning the green link:
I assume the vertical-line-text would probably interfere more often with the xticks when it is put directly under the line then if it stays at the side.
I tried pushing the text a bit up so it is aligned with the x-axis. How would you like that?

Reply:
can you post a screenshot?

Reply:
![grafik](https://user-images.githubusercontent.com/46245704/120647248-af3d0c80-c47a-11eb-9fc1-b13fb6f7117d.png)

Reply:
Just for the max. Zoom-Level this would exceed the available space. Maybe we could set a limit there ? (#97dfa0a)